File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        st, end = 0, len(nums) - 1
        while st < end:
            mid = st + (end - st)//2

            if nums[mid] > nums[end]:
                st = mid + 1
            else:
                end = mid
        
        return nums[st]

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int count = 0;
        int prefixSum = 0;
        unordered_map<int, int> map;
        map[0] = 1;

        for (int num: nums) {
            prefixSum += num;
            int diff = prefixSum - k;

            if (map.find(diff) != map.end()) {
                count += map[diff];
            }

            if (map.find(prefixSum) == map.end()) {
                map[prefixSum] = 0;
            }

            map[prefixSum]++;
        }

        return count;
    }
};